tests/opentelemetry-tests.bats

```diff
@@ -31,14 +31,53 @@ setup() {
 
     # Setup Kubewarden
     helm_in kubewarden-controller --reuse-values --values $RESOURCES_DIR/opentelemetry-kw-telemetry-values.yaml
-    helm_in kubewarden-defaults --reuse-values 
+    helm_in kubewarden-defaults --reuse-values --set "recommendedPolicies.enabled=True"
+
 }
 
 @test "[OpenTelemetry] Kubewarden containers have sidecar" {
     # Controller needs to be restarted to get sidecar
-    kubectl delete pod -n kubewarden -l app.kubernetes.io/name=kubewarden-controller
     wait_pods -n kubewarden
 
     # Check all pods have sidecar (otc-container) - might take a minute to start
     retry "kubectl get pods -n kubewarden -o json | jq -e '[.items[].spec.containers[1].name == \"otc-container\"] | all'"
 }
+
+@test "[OpenTelemetry] Policy server service has the metrics ports" {
+	retry "kubectl get services -n kubewarden  policy-server-default -o json | jq -e '[.spec.ports[].name == \"metrics\"] | any '"
+}
+
+@test "[OpenTelemetry] Controller service has the metrics ports" {
+	retry "kubectl get services -n kubewarden kubewarden-controller-metrics-service -o json | jq -e '[.spec.ports[].name == \"metrics\"] | any '"
+}
+
+@test "[OpenTelemetry] Policy server metrics should be available" {
+    # Generate metric data
+    kubectl run pod-privileged --image=registry.k8s.io/pause --privileged
+    retry "[ \$(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 10 ] || exit 1"
```

Comment:
Please double check this works, I wrote `retry` function for retrying simple commands, this is quite complex :)
Also retry function should fail by itself after 5 minutes (10*30s).

Reply:
Yes, I agree. 

I've tested this command with different parameters to ensure that it is validating the metrics. For example, I've tested the test in the command with the following values:

- `-gt 10` :  test passed as expected. Because the metrics usually have 40 lines or more
- `-gt 1000` : as expected the test failed after trying multiple times. 

Sharing the tests results:

```console
$ grep -rn "wc" tests/opentelemetry-tests.bats   
57:    retry "[ \$(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 10 ] || exit 1"
61:    retry "[ \$(kubectl run get-controller-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent kubewarden-controller-metrics-service.kubewarden.svc.cluster.local:8080/metrics | wc -l) -gt 1 ] || exit 1"


opentelemetry-tests.bats
 ✓ [OpenTelemetry] Install OpenTelemetry, Prometheus, Jaeger [timeout: 5m]
 ✓ [OpenTelemetry] Kubewarden containers have sidecar [timeout: 5m]
 ✓ [OpenTelemetry] Policy server service has the metrics ports [timeout: 5m]
 ✓ [OpenTelemetry] Controller service has the metrics ports [timeout: 5m]
 ✓ [OpenTelemetry] Policy server metrics should be available [timeout: 5m]
 ✓ [OpenTelemetry] Controller metrics should be available [timeout: 5m]
 ✓ [OpenTelemetry] User should be able to disable telemetry [timeout: 5m]
 ✓ [OpenTelemetry] Kubewarden containers have no sidecar [timeout: 5m]
 ✓ [OpenTelemetry] Policy server service has no metrics ports [timeout: 5m]
 ✓ [OpenTelemetry] Controller service has no metrics ports [timeout: 5m]

10 tests, 0 failures

```

```console

grep -rn "wc" tests/opentelemetry-tests.bats
57:    retry "[ \$(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1"
61:    retry "[ \$(kubectl run get-controller-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent kubewarden-controller-metrics-service.kubewarden.svc.cluster.local:8080/metrics | wc -l) -gt 1 ] || exit 1"

opentelemetry-tests.bats
 ✓ [OpenTelemetry] Install OpenTelemetry, Prometheus, Jaeger [timeout: 5m]
 ✓ [OpenTelemetry] Kubewarden containers have sidecar [timeout: 5m]
 ✓ [OpenTelemetry] Policy server service has the metrics ports [timeout: 5m]
 ✓ [OpenTelemetry] Controller service has the metrics ports [timeout: 5m]
 ✗ [OpenTelemetry] Policy server metrics should be available [timeout: 5m]
   (from function `retry' in file tests/common.bash, line 33,
    in test file tests/opentelemetry-tests.bats, line 57)
     `retry "[ \$(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1"' failed
   pod/pod-privileged created
   RETRY #1: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   If you don't see a command prompt, try pressing enter.
   RETRY #2: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #3: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #4: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #5: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #6: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #7: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #8: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #9: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Error from server (AlreadyExists): pods "get-policy-server-metric" already exists
   RETRY #10: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
   Godot: [ $(kubectl run get-policy-server-metric -t -i --rm --wait --image curlimages/curl:8.00.1 --restart=Never -- --silent policy-server-default.kubewarden.svc.cluster.local:8080/metrics | wc -l ) -gt 100 ] || exit 1
 ✓ [OpenTelemetry] Controller metrics should be available [timeout: 5m]
 ✓ [OpenTelemetry] User should be able to disable telemetry [timeout: 5m]
 ✓ [OpenTelemetry] Kubewarden containers have no sidecar [timeout: 5m]
 ✓ [OpenTelemetry] Policy server service has no metrics ports [timeout: 5m]
 ✓ [OpenTelemetry] Controller service has no metrics ports [timeout: 5m]

10 tests, 1 failure


```